src/EPR.Calculator.API/Controllers/CalculatorController.cs

```diff
@@ -52,5 +55,31 @@ public IActionResult GetCalculatorRun()
             // TODO: Return the details of a particular run
             return new OkResult();
         }
+
+        [HttpGet]
+        [Route("calculatorRunByNameExist/{name}")]
+        public IActionResult GetCalculatorRunByName([FromRoute] string name)
+        {
+            if (!ModelState.IsValid)
+            {
+                return StatusCode(StatusCodes.Status400BadRequest, ModelState.Values.SelectMany(x => x.Errors));
+            }
+
+            var calculatorRun = _context.CalculatorRuns.Any(run => run.Name.ToLower() == name.ToLower());
+
+            if (calculatorRun == false)
+            {
+                return new ObjectResult("No data found for this calculator name") { StatusCode = StatusCodes.Status404NotFound };
+            }
+
+            try
+            {
+                return new ObjectResult(StatusCodes.Status200OK );
```

Comment:
Why are you placing your try and catch only in sending the response? 
try/catch cover all the code?